src/transformStateWithClones.js

```diff
@@ -6,8 +6,33 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const statesArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateClone, action.extraData);
+        statesArray.push({ ...stateClone });
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateClone[key];
+        }
+        statesArray.push({ ...stateClone });
+        break;
+      case 'clear':
+        for (const key in stateClone) {
+          delete stateClone[key];
+        }
+        statesArray.push({ ...stateClone });
+        break;
+    }
```

Comment:
[CODE STYLE] - don't mutate object or arrays - it will cause unexpected results later on. Instead of modifying `stateClone` inside the switch cases and then pushing it into `statesArray`, create a new clone of `stateClone` inside each case, apply the modifications to this new clone, and then push it to `statesArray`. This ensures that you have a fresh clone for each action, avoiding unintended side effects.